sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/NumberPatternShouldBeRegular.cs

```diff
@@ -150,5 +129,24 @@ private static string StripNumericPreAndSuffix(string numericToken)
 
             return numericToken;
         }
+
+        private static bool HasIrregularGroupLengths(int[] groupLengths)
+        {
+            if (groupLengths.Length < 2)
+            {
+                return false;
+            }
+
+            // the second group is assumed to be leading.
+            var groupLength = groupLengths[1];
+
+            // first should not be bigger, and the size per group should be at least 2.
```

Comment:
this comment is a bit harder to read because you say "first should not be bigger" but that's the second expression in the `OR` expression

```suggestion
            // we consider groups of 1 digit irregular
            // the first group should not be bigger than the rest
```